src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
> The reson I do not think it is a BC break is because the return value of Plugin::handleRequest is a Promise. And a Promise can throw an \Exception.

It's not hard coded, but it's specified in the docblock of the HttpAsyncClient : https://github.com/php-http/httplug/blob/master/src/HttpAsyncClient.php#L22

So for me handling `\Exception` here is a BC break according to the docblock, i made this PR https://github.com/php-http/httplug/pull/116 to have this on the code on not only the docblock

And like i said in https://github.com/php-http/client-common/issues/34 i'm not against this change however we **should** be consistent and also deal with the case where the resolved value of our promise is not a `ResponseInterface` if we agree to do this.